src/lib.rs

```diff
@@ -1051,14 +1083,22 @@ where
 
     let initial_width = options
         .width
-        .saturating_sub(core::display_width(options.initial_indent));
+        .saturating_sub(core::display_width_with_tab(options.initial_indent, options.tab_width));
     let subsequent_width = options
         .width
-        .saturating_sub(core::display_width(options.subsequent_indent));
+        .saturating_sub(core::display_width_with_tab(options.subsequent_indent, options.tab_width));
 
     let mut lines = Vec::new();
+
     for line in text.split('\n') {
-        let words = options.word_separator.find_words(line);
+
+        // We have to set the tab width for each of the words. So as to not
+        // change the public API (fn find_words), we do it here.
+        let words = options.word_separator.find_words(line).map(|mut w| {
```

Comment:
It would definitely be more ideal if find_words could perhaps be implemented in Options or something like that, so that we wouldn't have to .map the result (and risk re-allocations? To be honest, I don't know exactly if this would cause more allocations or what the memory usage effect of this would be), but I'd like to hear thoughts on this and if this should be changed to achieve the same result through a different method.

Reply:
> It would definitely be more ideal if find_words could perhaps be implemented in `Options` or something like that

I would like to keep `Options` a mostly dumb container of, well, options. It is supposed to be the single place where library users configure the behavior of the top-level functions such as `wrap` and `fill`.